### PR TITLE
Fix openlayers with typescript 2.3 #15943 

### PR DESCRIPTION
**Pull request by tcoopman**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/DefinitelyTyped/DefinitelyTyped/issues/15943#issuecomment-296960129
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dslint/dt.json" }`.

### CODE REVIEW

**Comment by osechet** [comment]
What are the consequences with typescript <2.3?

**Comment by tcoopman** [comment]
I've just ran the test with typescript 2.2.2 and they also pass. I've also tried it on my project where I had problems before with 2.3 and now both 2.2.2 and 2.3 work.
I haven't tried it with more versions and it might be a good idea that someone with some better understanding of typescript can have a look to see why these changes were needed for the 2.3 upgrade.

File: types/openlayers/openlayers-tests.ts [hunk 1]
```diff
@@ -608,10 +609,10 @@ writeOptions.featureProjection = projection;
 writeOptions.rightHanded = false;
 
 // Test functions
-feature = geojsonFormat.readFeature("json");
-feature = geojsonFormat.readFeature("json", readOptions);
-featureArray = geojsonFormat.readFeatures("json");
-featureArray = geojsonFormat.readFeatures("json", readOptions);
+featureFormat = geojsonFormat.readFeature("json");
```
Comment (tcoopman) [inline types/openlayers/openlayers-tests.ts]: These changes are not ok, because readFeature should return an ol.Feature, not an ol.format.Feature

**Comment by tcoopman** [comment]
I've just detected a problem with this change (see review).
I'm going to see if I can fix this.

**Comment by tcoopman** [comment]
I've created a bug report on typescript to check what is wrong: https://github.com/Microsoft/TypeScript/issues/15400

I'm going to close this PR because I don't think it is a good idea to merge it.

**Comment by Julzk** [comment]
[works with typescript 2.3.0](https://github.com/Julzk/DefinitelyTyped/commit/f071b30c9969948448c75becb8f38f9a95fd884c)

**Pull request closed without merging**